Feat/plot grouping

bimac opened the pull request.

mdmelin committed:
allow user specified grouping variable for plotter

bimac committed:
a few refinements

bimac committed:
Update online_plots.py

bimac committed:
Update online_plots.py

bimac committed:
Update online_plots.py

mdmelin committed:
Add new functions to plotter as new grouping variables come in

mdmelin reviewed (changes requested):
@bimac I made a few more changes [here](https://github.com/int-brain-lab/iblrig/compare/feat/plot-grouping...mdmelin:iblrig:feat/plot-grouping). For online plotting during acquisition, the FunctionWidget needs to dynamically add new functions as new values for the grouping variable come in, since these now aren't known ahead of time.

bimac committed:
ruff

bimac committed:
Update online_plots.py

bimac commented:
Oh, good point. Looks good to me!

I moved a few things around and renamed a few variables - CamelCase vs snake_case is a PITA when it comes to PyQt. Will do some more testing on the dev rig tomorrow ...

oliche reviewed (approved):
Is there a test for the online plots ? 
I suppose there isn't , if it is too much work it's probably not a priority as the incidence of this component failing is low compared to others components of the project !

bimac commented:
> Is there a test for the online plots ? I suppose there isn't , if it is too much work it's probably not a priority as the incidence of this component failing is low compared to others components of the project !

Yes, there is: https://github.com/int-brain-lab/iblrig/blob/iblrigv8dev/iblrig/test/test_online_plots.py
